From 70cf01415e0ce8ca0c5cf683420bfbe916812fd5 Mon Sep 17 00:00:00 2001
From: Sergei Shkonda <sergeish@marvell.com>
Date: Sun, 29 Jan 2017 15:00:36 +0200
Subject: [PATCH 1033/1345] dts: armada: add support for 38x interposer card

commit  c3b076c2c54a6427c659a59dd60070bafb81fbf8 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

    88F6811, 88F6821 interposer card
    Applicable for:
       * AC3X RD board

    The only difference with armada-385-amc is an ethernet
    device PHY id changed and nand-ecc-strength

Change-Id: I3140ba189286d1a41d73aa0c807d231407091569
Reviewed-on: http://vgitil04.il.marvell.com:8080/39575
Reviewed-by: Terry Zhou <bjzhou@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/armada-38x-interposer.dts |   73 +++++++++++++++++++++++++++
 1 file changed, 73 insertions(+)
 create mode 100644 arch/arm/boot/dts/armada-38x-interposer.dts

diff --git a/arch/arm/boot/dts/armada-38x-interposer.dts b/arch/arm/boot/dts/armada-38x-interposer.dts
new file mode 100644
index 0000000..f1e3610
--- /dev/null
+++ b/arch/arm/boot/dts/armada-38x-interposer.dts
@@ -0,0 +1,73 @@
+/*
+ * Device Tree file for Marvell INTERPOSER card
+ * 88F6811, 88F6821 interposer card
+ * Applicable for:
+ *    * AC3X RD board
+ *
+ * Copyright (C) 2017 Marvell
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is licensed under the terms of the GNU General Public
+ *     License version 2.  This program is licensed "as is" without
+ *     any warranty of any kind, whether express or implied.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "armada-385-amc.dts"
+
+/ {
+	model = "Marvell Armada 38x interposer card";
+
+	soc {
+		internal-regs {
+			flash@d0000 {
+				nand-ecc-strength = <4>;
+			};
+
+			ethernet@70000 {
+				status = "disable";
+			};
+			ethernet@30000 {
+				pinctrl-names = "default";
+				/*
+				 * The Reference Clock 0 is used to provide a
+				 * clock to the PHY
+				 */
+				pinctrl-0 = <&ge0_rgmii_pins>, <&ref_clk0_pins>;
+				status = "okay";
+				phy = <&phy0>;
+				phy-mode = "rgmii-id";
+			};
+			mdio@72004 {
+				phy0: ethernet-phy@1 {
+					reg = <0>;
+				};
+			};
+		};
+	};
+};
-- 
1.7.9.5

